File: 0801_0900/0868/0868.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = (int(x) for x in f.readline().split())
    p = [[int(x) for x in f.readline().split()] for i in range(n)]
    col = 0
    for i in range(1, n):
        if max(p[i]) > max(p[col]):
            col = i
    row = p[col].index(max(p[col]))
    p.sort(key = lambda x : x[row], reverse = True)
    q.write(str(p[n - 1][p[0].index(min(p[0]))]))

File: 0801_0900/0868/0868.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, *p, r = 0, col = 0, row = 0;

	fscanf(f, "%d%d", &n, &m);
	p = (int*)malloc(sizeof(int) * n * m);
	for (int i = 0, s = n * m; i < s; ++i)
	{
		fscanf(f, "%d", p + i);
		if (p[i] > p[r])
			r = i;
	}
	for (int i = 1, l = r % m; i < n; ++i)
		if (i != r / m && p[i * m + l] < p[col * m + l])
			col = i;
	for (int i = 1, l = r / m * m; i < m; ++i)
		if (p[l + i] < p[l + row])
			row = i;
	fprintf(q, "%d", p[col * m + row]);
	return 0;
}
